core/src/saros/net/stream/Socks5StreamService.java

```diff
@@ -99,714 +56,217 @@
-      throws XMPPException, IOException, InterruptedException {
-    String peer = request.getFrom();
+  // *********** IStreamService impl end
 
-    log.debug(prefix() + "receiving request from " + peer + ", " + verboseLocalProxyInfo());
+  @Override
+  public void incomingBytestreamRequest(final BytestreamRequest request) {
 
-    IByteStreamConnectionListener listener = connectionListener;
+    final String remoteAddress = request.getFrom();
 
-    if (listener == null) throw new IOException(this + " transport is not initialized");
+    log.debug("received request to establish a Socks5 bytestream to: " + remoteAddress);
 
-    String sessionID = request.getSessionID();
+    final IByteStreamConnectionListener currentConnectionListener;
+    final JID currentLocalAddress;
 
-    String connectionIdentifier = getConnectionIdentifier(sessionID);
+    synchronized (this) {
+      currentConnectionListener = connectionListener;
+      currentLocalAddress = localAddress;
+    }
 
-    if (connectionIdentifier == null) {
-      log.warn(
-          "rejecting request from " + peer + " , no connection identifier found: " + sessionID);
+    if (currentConnectionListener == null || currentLocalAddress == null) {
+      log.warn("service is not initialized - rejecting request from: " + remoteAddress);
       request.reject();
-      return null;
+      return;
     }
 
-    assert sessionID != null;
-
-    // start to establish response
-    Future<Socks5BytestreamSession> responseFuture =
-        futureToEstablishResponseSession(sessionID, peer);
-
-    Socks5BytestreamSession inSession = null;
+    final Socks5BytestreamSession session;
 
     try {
+      ((Socks5BytestreamRequest) request).setTotalConnectTimeout(TOTAL_CONNECT_TIMEOUT);
+      session = (Socks5BytestreamSession) request.accept();
+    } catch (InterruptedException e) {
+      log.warn("interrupted while accepting request from: " + remoteAddress);
 
-      inSession = (Socks5BytestreamSession) request.accept();
-
-      if (inSession.isDirect()) {
-        waitToCloseResponse(responseFuture);
-        configureSocks5Socket(inSession);
-
-        return new BinaryChannelConnection(
-            localAddress,
-            new JID(peer),
-            connectionIdentifier,
-            new XMPPByteStreamAdapter(inSession),
-            StreamMode.SOCKS5_DIRECT,
-            listener);
-      } else {
-        log.debug(prefix() + "incoming connection is mediated.");
-      }
-
-    } catch (Exception e) {
-      log.warn(
-          prefix()
-              + "Couldn't accept request but still trying to establish a response connection: "
-              + e.getMessage());
+      /* This is called by Smack so avoid interrupting the thread again as nobody knows what will happen. */
+      // Thread.currentThread().interrupt();
+      return;
+    } catch (XMPPException e) {
+      log.error("failed to accept request from from: " + remoteAddress, e);
```

Comment:
```suggestion
      log.error("failed to accept request from from: {}", remoteAddress, e);
```

Reply:
Isnt this swallowing the exception ?

Reply:
The exception handling is not very well documented, but the default exception handling is described in [the layouts documentation (see `alwaysWriteExceptions`)](https://logging.apache.org/log4j/2.x/manual/layouts.html#Pattern_Layout):
> exceptions are always written even if the pattern contains no exception conversions. This means that if you do not include a way to output exceptions in your pattern, the default exception formatter will be added to the end of the pattern. Setting this to false disables this behavior and allows you to exclude exceptions from your pattern output.